sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/parser/SqlShowDatabases.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.beam.vendor.calcite.v1_40_0.org.apache.calcite.sql.SqlUtil;
+import org.apache.beam.vendor.calcite.v1_40_0.org.apache.calcite.sql.parser.SqlParserPos;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class SqlShowDatabases extends SqlCall implements BeamSqlParser.ExecutableStatement {
+  private static final SqlOperator OPERATOR =
+      new SqlSpecialOperator("SHOW DATABASES", SqlKind.OTHER_DDL);
+
+  private final boolean showCurrentOnly;
+  private final @Nullable SqlIdentifier catalogName;
+  private final @Nullable SqlNode regex;
+
+  public SqlShowDatabases(
+      SqlParserPos pos,
+      boolean showCurrentOnly,
+      @Nullable SqlIdentifier catalogName,
+      @Nullable SqlNode regex) {
+    super(pos);
+    this.showCurrentOnly = showCurrentOnly;
+    this.catalogName = catalogName;
+    this.regex = regex;
+  }
+
+  @Override
+  public List<SqlNode> getOperandList() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public SqlOperator getOperator() {
+    return OPERATOR;
+  }
+
+  @Override
+  public void execute(CalcitePrepare.Context context) {
+    Schema schema = SqlDdlNodes.schema(context, true).schema;
+
+    if (!(schema instanceof CatalogManagerSchema)) {
+      throw SqlUtil.newContextException(
+          pos,
+          RESOURCE.internal(
+              "Attempting to execute 'SHOW DATABASES' with unexpected Calcite Schema of type "
+                  + schema.getClass()));
+    }
+
+    CatalogSchema catalogSchema;
+    if (catalogName != null) {
+      String catalog = catalogName.toString();
+      catalogSchema = ((CatalogManagerSchema) schema).getCatalogSchema(catalog);
+    } else {
+      catalogSchema = ((CatalogManagerSchema) schema).getCurrentCatalogSchema();
+    }
+
+    if (showCurrentOnly) {
+      @Nullable BeamCalciteSchema currentDatabase = catalogSchema.getCurrentDatabaseSchema();
+      String output =
+          currentDatabase == null ? "No database is currently set" : currentDatabase.name();
+      System.out.println(output);
+      return;
+    }
+    Collection<String> databases = catalogSchema.databases();
+    print(databases, catalogSchema.getCatalog().name(), SqlDdlNodes.getString(regex));
+  }
+
+  private static void print(Collection<String> databases, String path, @Nullable String pattern) {
+    SqlFunctions.LikeFunction calciteLike = new SqlFunctions.LikeFunction();
+
+    final String headerName = "Databases in " + path;
+    final String separatorChar = "-";
+
+    int nameWidth = headerName.length();
+
+    for (String dbName : databases) {
+      nameWidth = Math.max(nameWidth, dbName.length());
+    }
+
+    nameWidth += 2;
+    String format = "| %-" + nameWidth + "s |%n";
+
+    int separatorWidth = nameWidth + 2;
+    String separator =
+        String.format(
+            "+" + new String(new char[separatorWidth]).replace("\0", separatorChar) + "+%n");
+
+    System.out.printf(separator);
```

Comment:
I'm not super familiar with sqlline but is this really the right way to implement SHOW? I would have expected it to maybe have a way that the command should return the string to be displayed rather than this stack-ignoring printf.